src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/entity/CsipRecord.kt

```diff
@@ -75,6 +83,12 @@ data class CsipRecord(
 
   fun referral() = referral
 
+  @OneToMany(
+    mappedBy = "csipRecord",
+    fetch = FetchType.LAZY,
+    cascade = [CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REMOVE],
+  )
+  var contributoryFactors: Collection<ContributoryFactor> = emptyList()
```

Comment:
These are really a child collection of the referral, not the record. Can they be moved to that level?

Reply:
This is what I struggled with, with the shared key :(